packages/flutter_riverpod/lib/src/consumer.dart

```diff
@@ -4,126 +4,17 @@ import 'package:meta/meta.dart';
 import 'internals.dart';
 
 /// An object that allows widgets to interact with providers.
-abstract class WidgetRef {
-  /// Returns the value exposed by a provider and rebuild the widget when that
-  /// value changes.
-  ///
-  /// See also:
-  ///
-  /// - [ProviderBase.select], which allows a widget to filter rebuilds by
-  ///   observing only the properties.
-  /// - [listen], to react to changes on a provider, such as for showing modals.
+abstract class WidgetRef implements Ref {
+  @override
   T watch<T>(ProviderListenable<T> provider);
 
-  /// Listen to a provider and call `listener` whenever its value changes.
-  ///
-  /// This is useful for showing modals or other imperative logic.
-  void listen<T>(
+  @override
+  RemoveListener listen<T>(
     ProviderListenable<T> provider,
     void Function(T? previous, T next) listener, {
+    bool fireImmediately,
```

Comment:
The prototype of `listen` should not change. `fireImmediately` is not supported, and there is no purpose in exposing the subscription